compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTileAndDistribute.cpp

```diff
@@ -70,16 +71,27 @@ static void populateTilingToInvocationPatterns(
                            .setTileSizeComputationFunction(computeFn)
                            .setDistributionOptions(distributionOptions);
 
-  MLIRContext *context = patterns.getContext();
+  MLIRContext *context = funcOp.getContext();
+  IRRewriter rewriter(context);
   auto marker = StringAttr::get(context, getTileReductionMarker());
   auto filter = IREE::LinalgExt::LinalgTransformationFilter(
-                    ArrayRef<StringAttr>(), marker)
-                    .setMatchByDefault();
+      ArrayRef<StringAttr>(), marker);
+
+  SmallVector<TilingInterface> candidates;
+  funcOp.walk([&](TilingInterface op) { candidates.push_back(op); });
+
+  for (auto op : candidates) {
+    FailureOr<IREETilingResult> res =
+        tileDispatchUsingSCFFopOp(rewriter, op, tilingOptions);
+    if (failed(res)) {
+      return failure();
+    }
+    for (auto tiledOp : res->tiledOps) {
+      filter.replaceLinalgTransformationFilter(rewriter, tiledOp);
+    }
+  }
 
-  patterns.add<IREE::LinalgExt::LinalgTilingPattern>(context, tilingOptions,
-                                                     filter);
-  patterns.add<IREE::LinalgExt::TilingInterfaceTilingPattern>(
```

Comment:
This pattern is a filter-based wrapper for tiling TilingInterface op. There are two options for tile+distribute linalg and linalg ops:

1. Use the `tileDispatchUsingSCFFopOp` method in `Codegen/Common/`.
2. Use the `tileInterfaceOp` from [LinalgExt/](https://github.com/openxla/iree/blob/682997301559c2ef7760bd8c3a59e6b69a0b4d23/llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/Tiling.cpp#L189-L190).

The goal is to remove LinalgExt legacy codes, so I think (1) is better.

Reply:
+1